diff --git a/app/controllers/application_controller.rb b/app/controllers/application_controller.rb
index e83e08989..f10f6de6c 100644
--- a/app/controllers/application_controller.rb
+++ b/app/controllers/application_controller.rb
@@ -689,8 +689,8 @@ class ApplicationController < ActionController::Base
     limit = options[:limit].to_i
     if limit < 1
       limit = 25
-    elsif limit > 100
-      limit = 100
+    elsif limit > 400
+      limit = 400
     end
     if offset.nil? && options[:page].present?
       offset = (options[:page].to_i - 1) * limit
