--- a/Gemfile
+++ b/Gemfile
@@ -98,7 +98,7 @@ group :test do
   gem "ffi", platforms: [:mingw, :x64_mingw, :mswin]
   # For running system tests
   # TODO: Remove version specification once Capybara supports Puma 6
-  gem 'puma', '< 6.0.0'
+  gem 'puma'
   gem 'capybara', (Gem.ruby_version < Gem::Version.new('2.6.0') ? '~> 3.35.3' : '~> 3.36.0')
   gem "selenium-webdriver", "~> 3.142.7"
   gem 'webdrivers', '4.6.1', require: false
