diff --git a/plugins/redmine_dmsf/app/controllers/dmsf_upload_controller.rb b/plugins/redmine_dmsf/app/controllers/dmsf_upload_controller.rb
index 401402ef..981893b3 100644
--- a/plugins/redmine_dmsf/app/controllers/dmsf_upload_controller.rb
+++ b/plugins/redmine_dmsf/app/controllers/dmsf_upload_controller.rb
@@ -73,7 +73,7 @@ class DmsfUploadController < ApplicationController
       Attachment.skip_callback(:commit, :after, :reuse_existing_file_if_possible, raise: false)
       saved = @attachment.save
     ensure
-      Attachment.set_callback(:commit, :after, :reuse_existing_file_if_possible)
+      Attachment.set_callback(:commit, :after, :reuse_existing_file_if_possible, on: :create)
     end

     respond_to do |format|
