diff --git a/app/models/query.rb b/app/models/query.rb
index 7695bc165..209e68e01 100644
--- a/app/models/query.rb
+++ b/app/models/query.rb
@@ -1160,9 +1160,10 @@ class Query < ActiveRecord::Base
       where = "(#{where}) AND #{db_table}.#{db_field} <> ''"
     end
     "#{queried_table_name}.#{customized_key} #{not_in} IN (" \
-      "SELECT #{customized_class.table_name}.id FROM #{customized_class.table_name}" \
+      "SELECT cc_sub.id FROM #{customized_class.table_name} AS cc_sub" \
       " LEFT OUTER JOIN #{db_table} ON #{db_table}.customized_type='#{customized_class}'" \
-      " AND #{db_table}.customized_id=#{customized_class.table_name}.id" \
+      " AND #{db_table}.customized_id=cc_sub.id" \
+      " AND #{queried_table_name}.id=cc_sub.id " \
       " AND #{db_table}.custom_field_id=#{custom_field_id}" \
       " WHERE (#{where}) AND (#{filter[:field].visibility_by_project_condition}))"
   end
